tt_metal/hostdevcommon/api/hostdevcommon/flags.hpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <type_traits>
+
+template <typename E>
+struct Flags {
+    static_assert(std::is_enum_v<E>, "Flags<E> requires E to be an enum.");
+
+    using Underlying = std::underlying_type_t<E>;
+
+    constexpr Flags() noexcept : bits_(0) {}
+    constexpr Flags(E single) noexcept : bits_(static_cast<Underlying>(single)) {}
+    constexpr Flags(E a, E b) noexcept : bits_(static_cast<Underlying>(a) | static_cast<Underlying>(b)) {}
+    constexpr explicit Flags(Underlying bits) noexcept : bits_(bits) {}
+
+    // Bitwise OR a single enum
+    constexpr Flags operator|(E rhs) const noexcept { return Flags(bits_ | static_cast<Underlying>(rhs)); }
+    // Bitwise OR another Flags
+    constexpr Flags operator|(Flags rhs) const noexcept { return Flags(bits_ | rhs.bits_); }
+
+    // ... similarly, operator&, operator^, operator~ if you like ...
+    constexpr Flags operator&(E rhs) const noexcept { return Flags(bits_ & static_cast<Underlying>(rhs)); }
+    constexpr bool test(E single) const noexcept {
+        return (bits_ & static_cast<Underlying>(single)) == static_cast<Underlying>(single);
+    }
```

Comment:
While reviewing your usage of `.test`, I noticed that it only checks whether a single bit is set. This behavior is equivalent to `std::set::contains()` as described here: [https://en.cppreference.com/w/cpp/container/set/contains](https://en.cppreference.com/w/cpp/container/set/contains).

Since `test` is a relatively vague name, I recommend renaming it to `.contains` for clarity and semantic alignment with the standard library.

```suggestion
    constexpr bool contains(E value) const noexcept { return (bits_ & static_cast<Underlying>(value)) != 0; }
```